internal/builders/go/pkg/build_test.go

```diff
@@ -1023,15 +1005,17 @@ func Test_generateCommand(t *testing.T) {
 				},
 			}
 
-			for _, cfg := range cfgs {
+			for i := range cfgs {
+				cfg := cfgs[i]
```

Comment:
This was reported by `gosec`'s rule `G601`.  The potential problem arises when we take the pointer of the iteration variable, as we do when passing it to `fromConfig`. The iteration variable is reused in each iteration. When someone remembers the pointer they will see changing values. 

I don't think the current code is unsafe as it is, but someone could make changes and store that pointer for later use leading to unexpected behavior. 

[This stackoverflow post provides more details and potential solutions.](https://stackoverflow.com/questions/62446118/implicit-memory-aliasing-in-for-loop)

I'd prefer indexing, but a local copy would also work.

Reply:
Yeah, I'm familiar with this issue. The problem I have is that it's sometimes more desirable to have a local copy since the original slice can't be modified in unexpected ways.

I think this is ok, but maybe it can be avoided by making the `cfgs` a list of pointers? Not sure, `gosec` may still complain anyway though.